xarray/core/dtypes.py

```diff
@@ -61,7 +62,12 @@ def maybe_promote(dtype: np.dtype) -> tuple[np.dtype, Any]:
     # N.B. these casting rules should match pandas
     dtype_: np.typing.DTypeLike
     fill_value: Any
-    if isdtype(dtype, "real floating"):
+    if HAS_STRING_DTYPE and np.issubdtype(dtype, np.dtypes.StringDType()):  # type: ignore[attr-defined]
```

Comment:
```suggestion
    if HAS_STRING_DTYPE and np.issubdtype(dtype, np.dtypes.StringDType()):  # type: ignore[attr-defined]
        # the check for HAS_STRING_DTYPE should be removed once we can drop support for `numpy<2.0`
```

Reply:
that should be fine, though I guess if we follow that logic we don't actually have to do anything: since the comment in the block you quoted is already a marker to ensure that we remove it when cleaning up.

Reply:
Correct, I think this will all be cleaned up at the same time. `HAS_STRING_DTYPE` is synonymous with having numpy 2.